### ISSUE TITLE
Issue submitting Week 4 - Recover solution

### ISSUE THREAD

**Issue by walterpalladino** [cs50/check50#294]
When I run valgrind I see no errors:

recover/ $ valgrind --show-leak-kinds=all -- ./recover card.raw
==26893== Memcheck, a memory error detector
==26893== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==26893== Using Valgrind-3.18.1 and LibVEX; rerun with -h for copyright info
==26893== Command: ./recover card.raw
==26893== 
==26893== 
==26893== HEAP SUMMARY:
==26893==     in use at exit: 0 bytes in 0 blocks
==26893==   total heap usage: 104 allocs, 104 frees, 233,736 bytes allocated
==26893== 
==26893== All heap blocks were freed -- no leaks are possible
==26893== 
==26893== For lists of detected and suppressed errors, rerun with: -s
==26893== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)

But I get this report when submitted :
 
:( program is free of memory errors
Cause
timed out while waiting for program to exit
Log
running valgrind --show-leak-kinds=all --xml=yes --xml-file=/tmp/tmpq8566_34 -- ./recover card.raw...

![image](https://user-images.githubusercontent.com/1933925/203635520-5551348c-0852-4348-b045-be99ac27b354.png)

**Comment by rongxin-liu** [comment]
On what operating systems and platforms did you run your code with valgrind?

**Comment by walterpalladino** [comment]
I'm using the web environment

**Comment by rongxin-liu** [comment]
If you run `check50 --local cs50/problems/2022/x/recover` in your recover pset folder, are you getting the same error?

**Comment by walterpalladino** [comment]
Yes, I get the same error but it generates the report as an html file

recover/ $ check50 --local cs50/problems/2022/x/recover
Checking..............................................
Results for cs50/problems/2022/x/recover generated by check50 v3.3.7
:) recover.c exists.
:) recover.c compiles.
:) handles lack of forensic image
:) recovers 000.jpg correctly
:) recovers middle images correctly
:) recovers 049.jpg correctly
:( program is free of memory errors
    timed out while waiting for program to exit
To see the results in your browser go to file:///tmp/tmpdophitdw.html

**Comment by rongxin-liu** [comment]
> The report says nothing about performance and to be hones I don-t this this should be evaluated at this moment (week 4).

I'm sorry, your code did not finish running in 10 seconds with valgrind. Please double-check your implementation regarding performance since lecture 3 already talked about efficiency and running time.

**Comment by curiouskiwi** [comment]
Just as a hint
> Moreover, rather than read my memory card’s bytes one at a time, you can read 512 of them at a time into a buffer for efficiency’s sake. Thanks to FAT, you can trust that JPEGs’ signatures will be “block-aligned.” That is, you need only look for those signatures **in a block’s first four bytes.**
